definitions/procedures/backup/snapshot/mount_candlepin_db.rb

```diff
@@ -0,0 +1,46 @@
+require 'procedures/backup/snapshot/mount_base'
+module Procedures::Backup
+  module Snapshot
+    class MountCandlepinDB < MountBase
+      metadata do
+        description 'Create and mount snapshot of Candlepin DB'
+        tags :backup
+        label :backup_snapshot_mount_candlepin_db
+        for_feature :candlepin_database
+        preparation_steps { Checks::Candlepin::DBUp.new unless feature(:candlepin_database).local? }
+        MountBase.common_params(self)
+        param :backup_dir, 'Directory where to backup to'
+      end
+
+      def run
+        if feature(:candlepin_database).local?
+          mount_snapshot
+        else
+          dump_candlepin
+        end
+      end
+
+      private
+
+      def dump_candlepin
+        puts 'LV snapshots are not supported for remote databases. Doing postgres dump instead... '
+        with_spinner('Getting Candlepin DB dump') do
+          feature(:candlepin_database).dump_db(File.join(@backup_dir, 'candlepin.dump'))
+        end
+      end
+
+      def mount_snapshot
+        if Dir.entries(mount_location('pgsql')).empty?
```

Comment:
```
E, [2018-04-06 19:04:15+0200 #29591] ERROR -- : No such file or directory - /var/
snap/pgsql (Errno::ENOENT)
/usr/share/gems/gems/foreman_maintain-0.1.5/definitions/procedures/backup/snapshot/mount_foreman_db.rb:33:in `open'
```
it checks the directory /var/snap/pgsql before MountBase::mount_snapshot creates it

and will it work if MountBase::mount_snapshot is overwritten by MountCandlepinDB::mount_snapshot?

Reply:
It checks the mount point just to know if it was already created by the other DB backup (candlepin vs foreman). I'll fix the condition.

@alda519, the code assumes `/var` exist. How come it is not present on your system? Is it wrong assumption?

Reply:
I've added check that the mount point does exist.

Reply:
I see now what is going on there. Thanks! All issues you've reported should be fixed now.